### DISCUSSION TITLE
"Only root package requirements can receive temporary constraints" problem

### DISCUSSION THREAD

**Discussion by ghost** [composer/composer#10984] [Q&A / Support] [answered]
Hi,
I wonder why composer refuses to update/downgrade a non-root package to the fixed version. I need to **down**grade packages from some vendor. Unfortunately it's impossible to run `composer update vendor/*` because composer complies about conflicts. Using `with-dependencies` and `--with-all-dependencies` doesn't work either because some other packages are still in conflicts and a message `other-vendor/xyz x.y.z requires vendor/package (..) but these were not loaded, likely because it conflicts with another require` shows up.

What I came up with is to run just `composer update` (after downgrading version constraints next to the desired packages in my composer.json of course) **and then** downgrade all these other packages that were updated as a side effect to the versions they where at before. However a limitation mentioned in Discussion title shows up. 
**My question is**: If I have a non-root package is required by another package at constraint `^1.0` then why I can't downgrade this non-root package to the specific version which still covers `^1.0` constraint?

**Comment by stof** [comment, marked as answer]
The restriction to apply them only on root requirements is already removed in the upcoming 2.4 version of Composer (you can try with with the snapshot builds)
